### PR TITLE
Projections change representation

### PR DESCRIPTION
**Pull request by mateuszbaran**
This PR fixes rotations manifold so that projections actually change the representation of tangent vectors. Also a couple of related changes.

### CODE REVIEW

File: src/manifolds/Circle.jl [hunk 1]
```diff
@@ -105,6 +105,19 @@ distance(::Circle{ℝ}, p::Real, q::Real) = abs(sym_rem(p - q))
...
+
+Embed a point `p` on [`Circle`](@ref) `M` in the ambient space. It returns `p`.
+"""
+embed(::Circle, p) = p
+@doc raw"""
+    embed(M::Circle, p, X)
+
+Embed a tangent vector `X` at `p` on [`Circle`](@ref) `M` in the ambient space. It returns `X`.
+"""
+embed(::Circle, p, X) = X
```
Comment (kellertuer) [inline src/manifolds/Circle.jl:119]: I thought this was the default, but if it is not, then sure this line is correct.

**Review by kellertuer — COMMENTED** [review]
Besides one small question this looks fine – thanks for spotting my bugs :)

File: src/manifolds/Circle.jl (hunk 1 above)
Comment (mateuszbaran) [reply src/manifolds/Circle.jl:119]: No, the default is to call the mutating `embed!` and it fails for `Circle` when it uses immutable numbers.

Comment (kellertuer) [reply src/manifolds/Circle.jl:119]: ah, right!

**Review by kellertuer — APPROVED** [review]

**Pull request merged**